### PR TITLE
Add SBD dump gatherer documentation

### PR DESCRIPTION
**Pull request by nelsonkopliku**
As per the title, added doc for the SBD dump gatherer https://github.com/trento-project/agent/pull/156

### CODE REVIEW

File: guides/gatherers.md [hunk 1]
```diff
@@ -197,6 +198,54 @@ facts:
...
+
+Specification examples:
+
+```yaml
+facts:
+  - name: sbd_devices_dump
+    gatherer: sbd_dump
+
+  - name: sbd_devices_dump
+    gatherer: whatever_argument # ignored
```
Comment (rtorrero) [inline guides/gatherers.md:244]: I think you mean here:
```
gatherer: sbd_dump
argument: whatever_argument # ignored
```

**Review by rtorrero — APPROVED** [review]
LGTM, one minor comment :+1:

File: guides/gatherers.md (hunk 1 above)
Comment (nelsonkopliku) [reply guides/gatherers.md]: Yep thanks!

File: guides/gatherers.md [hunk 2]
```diff
@@ -197,6 +198,55 @@ facts:
 
 For extra information refer to [trento-project/agent/../gatherers/sbd_test.go](https://github.com/trento-project/agent/blob/main/internal/factsengine/gatherers/sbd_test.go)
 
+### sbd_dump
+
+This gatherer allows accessing the dumped information about all the configured devices on target node.
+Simply executes `sbd -d <device> dump` for all the devices in `/etc/sysconfig/sbd` and aggregates results in only one fact.
```
Comment (dottorblaster) [inline guides/gatherers.md:204]:
```suggestion
It executes `sbd -d <device> dump` for all the devices in `/etc/sysconfig/sbd` and aggregates results in only one fact.
```

File: guides/gatherers.md [hunk 3]
```diff
@@ -197,6 +198,55 @@ facts:
 
 For extra information refer to [trento-project/agent/../gatherers/sbd_test.go](https://github.com/trento-project/agent/blob/main/internal/factsengine/gatherers/sbd_test.go)
 
+### sbd_dump
+
+This gatherer allows accessing the dumped information about all the configured devices on target node.
```
Comment (dottorblaster) [inline guides/gatherers.md:203]:
```suggestion
This gatherer allows accessing the dumped data about all the configured devices on target nodes.
```

File: guides/gatherers.md [hunk 4]
```diff
@@ -197,6 +198,55 @@ facts:
...
+Simply executes `sbd -d <device> dump` for all the devices in `/etc/sysconfig/sbd` and aggregates results in only one fact.
+
+Note that:
+- no arguments are required and any provided argument would just be ignored.
+- if dumping any of the devices fails, the fact would be a fact error
```
Comment (dottorblaster) [inline guides/gatherers.md:208]:
```suggestion
- if any dump fails, the return value will be a fact error
```

**Review by dottorblaster — CHANGES_REQUESTED** [review]
Just some tiny changes then we're good to merge

File: guides/gatherers.md (hunk 2 above)
Comment (nelsonkopliku) [reply guides/gatherers.md]: thanks, changed!

File: guides/gatherers.md (hunk 3 above)
Comment (nelsonkopliku) [reply guides/gatherers.md]: thanks, changed!

File: guides/gatherers.md (hunk 4 above)
Comment (nelsonkopliku) [reply guides/gatherers.md]: thanks, changed!

**Review by dottorblaster — APPROVED** [review]
LGTM

File: guides/gatherers.md [hunk 5]
```diff
@@ -197,6 +198,55 @@ facts:
 
 For extra information refer to [trento-project/agent/../gatherers/sbd_test.go](https://github.com/trento-project/agent/blob/main/internal/factsengine/gatherers/sbd_test.go)
 
+### sbd_dump
+
+This gatherer allows accessing the dumped data about all the configured devices on target nodes.
```
Comment (arbulu89) [inline guides/gatherers.md:203]: I would rephrase this as it is not really accurate:
```
This gatherer allows accessing the sbd dump command output data.

It executes the `sbd -d <device> dump` command in all devices configured in the `SBD_DEVICES` field on `/etc/sysconfig/sbd` and aggregates results in only one fact.
```

File: guides/gatherers.md [hunk 6]
```diff
@@ -197,6 +198,55 @@ facts:
...
+It executes `sbd -d <device> dump` for all the devices in `/etc/sysconfig/sbd` and aggregates results in only one fact.
+
+Note that:
+- no arguments are required and any provided argument would just be ignored.
+- if any dump fails, a fact error is returned
+
+```rust
```
Comment (arbulu89) [inline guides/gatherers.md:210]: Maybe put a header here saying that the "keys" are sanitized and this is how they can access the values in `rhai`

File: guides/gatherers.md [hunk 7]
```diff
@@ -197,6 +198,55 @@ facts:
...
+Specification examples:
+
+```yaml
+facts:
+  - name: sbd_devices_dump
+    gatherer: sbd_dump
+
+  - name: sbd_devices_dump
+    gatherer: sbd_dump
+    argument: whatever_argument # ignored
```
Comment (fabriziosestito) [inline guides/gatherers.md:245]: I would remove the ignored argument, as the gatherer does not accept arguments and this is something it shouldn't be present anyway

**Review by fabriziosestito — CHANGES_REQUESTED** [review]

File: guides/gatherers.md [hunk 8]
```diff
@@ -197,6 +198,55 @@ facts:
 
 For extra information refer to [trento-project/agent/../gatherers/sbd_test.go](https://github.com/trento-project/agent/blob/main/internal/factsengine/gatherers/sbd_test.go)
 
+### sbd_dump
+
+This gatherer allows accessing the dumped data about all the configured devices on target nodes.
+It executes `sbd -d <device> dump` for all the devices in `/etc/sysconfig/sbd` and aggregates results in only one fact.
+
+Note that:
+- no arguments are required and any provided argument would just be ignored.
```
[the remark this answers is not in the document]
Comment (nelsonkopliku) [reply guides/gatherers.md:207]: It was mainly for completeness, and yes, we can cut it short. Changing.

File: guides/gatherers.md (hunk 6 above)
Comment (nelsonkopliku) [reply guides/gatherers.md:213]: Added the line above
> Dumped keys (`Timeout (watchdog)`, `Timeout (msgwait)`, `Number of slots`, etc) are sanitized to simplify their access and usage in the expression language.

How does that work?

File: guides/gatherers.md [hunk 9]
```diff
@@ -197,6 +198,53 @@ facts:
 
 For extra information refer to [trento-project/agent/../gatherers/sbd_test.go](https://github.com/trento-project/agent/blob/main/internal/factsengine/gatherers/sbd_test.go)
 
+### sbd_dump
+
+This gatherer allows accessing the sbd dump command output data.
+
+It executes the `sbd -d <device> dump` command in all devices configured in the `SBD_DEVICES` field on `/etc/sysconfig/sbd` and aggregates results in only one fact.
```
Comment (arbulu89) [inline guides/gatherers.md:205]: Shit, my fault, the entry is `SBD_DEVICE` without the final `s`

**Review by arbulu89 — APPROVED** [review]
Thanks @nelsonkopliku 
Sorry for the typo in the `SBD_DEVICE` ...

**Comment by nelsonkopliku** [comment]
@fabriziosestito ok with the latest changes?

**Review by fabriziosestito — APPROVED** [review]
